editor-tool-autocomplete/editor-tool-autocomplete-api/src/main/java/org/xwiki/editor/tool/autocomplete/internal/AutoCompletionResource.java

```diff
@@ -353,11 +362,12 @@ private AutoCompletionMethodFinder getMethodFinder(String hint)
     protected VelocityContext getVelocityContext()
     {
         VelocityContext velocityContext = this.velocityManager.getVelocityContext();
+        XWikiContext context = this.xcontextProvider.get();
 
         // We add the "doc", "tdoc" and "sdoc" mappings since we don't get them from the Velocity Manager as they are
         // normally added based on the document passed in the request. However since we return the same method names
         // whatever the doc, we can manually add them.
-        XWikiDocument fakeDocument = createFakeXWikiDocument();
+        Document fakeDocument = createFakeXWikiDocument().newDocument(context);
```

Comment:
Hi Yann,

We should not use the Document class in java code. This is meant for scripting APIs only. In Java, we should only use the XWikiDocument class.

In your commit it's not explained why you need to call newDocument(context), could you explain?

Thanks